cmake/README.md

```diff
@@ -0,0 +1,76 @@
+# protovalidate-cc CMake Usage
+
+protovalidate-cc comes with a CMake build script. You can use this to embed
+protovalidate-cc into your existing CMake project or install protovalidate-cc
+system-wide.
+
+## Embedding
+
+protovalidate-cc requires C++17 or higher. Near the top of your CMakeLists,
+ensure you are setting the C++ standard:
+
+```
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+```
+
+You can use the following snippet to include protovalidate-cc.
+
+```
+FetchContent_Declare(
+    protovalidate_cc
+    GIT_REPOSITORY "https://github.com/bufbuild/protovalidate-cc.git"
+    GIT_TAG "v0.5.0"
+)
+FetchContent_MakeAvailable(protovalidate_cc)
+```
+
+When built with CMake, protovalidate-cc has the following external dependencies:
+
+- [Abseil](https://abseil.io/)
+- [Protocol Buffers](https://protobuf.dev/)
+- [re2](https://github.com/google/re2)
+- [ANTLR 4](https://www.antlr.org/)
+- [googleapis](https://github.com/googleapis/googleapis.git)
+- [cel-cpp](https://github.com/google/cel-cpp)
+- [protovalidate](https://github.com/bufbuild/protovalidate)
+
+Depending on your project setup, these can be sourced in different ways:
+
+- Abseil, Protocol Buffers and re2 can link to external copies. If you already
+  have Abseil, Protocol Buffers, or re2 added to your CMake project *prior to
+  the `FetchContent_MakeAvailable` for protovalidate-cc*, the existing targets
+  will be used. Please note that in this case, you need to ensure that the
+  versions of Abseil, Protocol Buffers, and re2 are supported by cel-cpp and
+  protovalidate-cc. The versions specified in
+  [`deps/shared_deps.json`](../deps/shared_deps.json) are known to compile and
+  function correctly.
```

Comment:
Just fyi - `deps/shared_deps.json` doesn't seem to specify a version of `re2`.

Reply:
True, that's not a shared dependency so it's not specified. I'll think about what we should do here. We can't add re2 to the list as it will cause Bazel to try to fetch it as well, which would be unnecessary.

Reply:
I linked elsewhere instead.